### PR TITLE
move `nomad init` outputs to go-bindata assets

### PR DESCRIPTION
**Pull request by tgross**
This PR is prep work for https://github.com/hashicorp/nomad/issues/6127

Having these files embedded as consts makes it harder to review and edit them (ex. no syntax highlighting or easy `hclfmt`), and adding two more files will compound that. We're already using a similar process for UI assets, so this doesn't seem like too much of a lift.

### CODE REVIEW

**Comment by tgross** [comment]
We were getting allocation failures in CI during linting, so I've added e444b37 in an effort to reduce the memory usage of the `make check` step.

File: .circleci/config.yml [hunk 1]
```diff
@@ -74,6 +74,7 @@ jobs:
   lint:
     executor: go
     environment:
+      <<: *COMMON_ENVS
```
Comment (endocrimes) [inline .circleci/config.yml]: We accidentally bring in a couple of other env vars here but they're safe :+1:

File: GNUmakefile [hunk 2]
```diff
@@ -217,6 +217,10 @@ proto:
 		protoc -I . -I ../../.. --go_out=plugins=grpc:. $$file; \
 	done
 
+.PHONY: generate-examples
+generate-examples: command/job_init.bindata_assetfs.go
```
Comment (endocrimes) [inline GNUmakefile]:
```suggestion
generate-examples: command/job_init.bindata_assetfs.go

```

Would be nice to just give ourselves a lil bit of whitespace here

File: command/assets/example.nomad [hunk 3]
```diff
@@ -0,0 +1,394 @@
+# There can only be a single job definition per file. This job is named
+# "example" so it will create a job with the ID and Name "example".
```
Comment (endocrimes) [inline command/assets/example.nomad]: could you run [`hclfmt`](https://github.com/fatih/hclfmt) over these as quick double check for consistent formatting??

File: GNUmakefile (hunk 2 above)
Comment (tgross) [reply GNUmakefile]: Sure thing.

File: command/assets/example.nomad (hunk 3 above)
Comment (tgross) [reply command/assets/example.nomad]: Just was looking at that and it looks like `hclfmt` is mangling a few of the multi-line comments in the bigger `example.nomad` file. I'm in the middle of trying to figure out why but once I've got it figured out I will do that.

Comment (tgross) [reply command/assets/example.nomad]: I think what I was seeing may be related to https://github.com/hashicorp/hcl/issues/253 but I've been able to fix it by ensuring that the comments following a top-level object are always followed by a non-comment line.

That is:
```diff
@@ -15,7 +15,7 @@ job "example" {
   # The "region" parameter specifies the region in which to execute the job.
   # If omitted, this inherits the default region name of "global".
   # region = "global"
-
+  #
   # The "datacenters" parameter specifies the list of datacenters which should
   # be considered when placing this task. This must be provided.
   datacenters = ["dc1"]

```

**Pull request merged**